### ISSUE TITLE
Test NCIP Toolkit with HTTPS, other security concerns

### ISSUE THREAD

**Issue by patrickzurek** [eXtensibleCatalog/xcncip2toolkit#115]
JIRA issue created by: rcook
Originally opened: 2012-06-27 12:36 PM

Issue body:

http://code.google.com/p/xcncip2toolkit/issues/detail?id=35

Notify [~jbodfish]

Reported by project member rcook@library.rochester.edu, Jul 28, 2011

Need to design an appropriate test and execute.

Address security concerns related to passing User ID and Authentication information via XML.

Also system review that no inappropriate data is written to logs, etc.

Note that this is a task, not that we have a known issue.

**Comment by patrickzurek** [comment]
JIRA Coment by user: rcook
JIRA Timestamp: 2012-08-29 11:11 AM

Comment body:

Do either you or [~bodfishj] think we have follow up work here?

**Comment by patrickzurek** [comment]
JIRA Comment by user: Patrick Zurek (patrickzurek)
JIRA Timestamp: 2012-09-05 05:42 PM

Comment body:

I tested HTTPS by creating a keystore and configuring Tomcat to use it per the instructions here (http://tomcat.apache.org/tomcat-6.0-doc/ssl-howto.html).  Tested by going to https://xc-zurek.carli.illinois.edu:8443/voyager (this may not be accessible off the UIUC campus -- I think the campus firewall is blocking port 8443).

Regarding logging of sensitive data, in LookupUser the user's barcode and last name were being written to the log files by the Apache HttpClient.  This is because the library logged the HTTP request string and the login credentials were included in the request URL.  These messages have been suppressed by changing the log level in the log4j.properties file to:

log4j.logger.httpclient.wire.header=WARN
log4j.logger.httpclient.wire.content=WARN
